rasa/nlu/tokenizers/convert_tokenizer.py

```diff
@@ -0,0 +1,145 @@
+
+
+class ConveRTTokenizer(WhitespaceTokenizer):
+
+    provides = [TOKENS_NAMES[attribute] for attribute in MESSAGE_ATTRIBUTES]
+
+    defaults = {
+        # Flag to check whether to split intents
+        "intent_tokenization_flag": False,
+        # Symbol on which intent should be split
+        "intent_split_symbol": "_",
+        # Text will be tokenized with case sensitive as default
+        "case_sensitive": True,
+    }
+
+    def __init__(self, component_config: Dict[Text, Any] = None) -> None:
+        """Construct a new tokenizer using the WhitespaceTokenizer framework."""
+
+        super().__init__(component_config)
+
+        self._load_tokenizer_params()
+
+    def _load_tokenizer_params(self):
+
+        # needed to load the ConveRT model
+        import tensorflow_text
+        import tensorflow_hub as tfhub
+
+        self.graph = tf.Graph()
+        model_url = "http://models.poly-ai.com/convert/v1/model.tar.gz"
+
+        with self.graph.as_default():
+            self.session = tf.Session()
+            self.module = tfhub.Module(model_url)
+
+            self.text_placeholder = tf.placeholder(dtype=tf.string, shape=[None])
+            self.tokenized = self.module(self.text_placeholder, signature="tokenize")
+
+            self.session.run(tf.tables_initializer())
+            self.session.run(tf.global_variables_initializer())
+
+    def train(
+        self, training_data: TrainingData, config: RasaNLUModelConfig, **kwargs: Any
+    ) -> None:
+        for example in training_data.training_examples:
+            for attribute in MESSAGE_ATTRIBUTES:
+                if example.get(attribute) is not None:
+                    example.set(
+                        TOKENS_NAMES[attribute],
+                        self.tokenize_using_convert(example.get(attribute), attribute),
+                    )
+
+    def process(self, message: Message, **kwargs: Any) -> None:
+        message.set(
+            TOKENS_NAMES[TEXT_ATTRIBUTE], self.tokenize_using_convert(message.text)
+        )
+
+    def _tokenize(self, sentence: Text) -> Any:
+        return self.session.run(
+            self.tokenized, feed_dict={self.text_placeholder: [sentence]}
+        )
+
+    def tokenize_using_convert(
+        self, text: Text, attribute: Text = TEXT_ATTRIBUTE
+    ) -> List[Token]:
+        """Tokenize the text using the ConveRT model.
+
+        ConveRT adds a special char in front of (some) words and splits words into
+        sub-words. To ensure the entity start and end values matches the token values,
+        tokenize the text first using the whitespace tokenizer. If individual tokens
+        are split up into multiple tokens, we make sure that the start end end value
+        of the first and last respective tokens stay the same.
+        """
+
+        # perform whitespace tokenization
+        tokens_in = self.tokenize(text, attribute)
+
+        # remove CLS token if present
+        if tokens_in[-1].text == CLS_TOKEN:
```

Comment:
in different algorithms we either check ` if attribute in [RESPONSE_ATTRIBUTE, TEXT_ATTRIBUTE]:` or `if tokens_in[-1].text == CLS_TOKEN:` I guess we should use the same if statement everywhere if possible